tests/test_errors.py

```diff
@@ -532,30 +532,31 @@ def finalize(self, cfg: Any) -> None:
     pytest.param(
         Expected(
             create=lambda: None,
-            op=lambda cfg: OmegaConf.structured(NotOptionalInt),
+            op=lambda _: OmegaConf.structured(NotOptionalInt),
             exception_type=ValidationError,
             msg="Non optional field cannot be assigned None",
-            object_type_str=None,
-            ref_type_str=None,
+            key="foo",
+            object_type=NotOptionalInt,
+            parent_node=lambda _: {},  # dummy parent
```

Comment:
why do you need to add those parent_node fields?

Reply:
This is because `OmegaConf.structured(...)` creates a `dummy_parent` node (in `_utils.py:get_dataclass_data` or `get_attr_data`).

Lower down in this test file we have
```python
assert ex.parent_node == expected.parent_node(cfg)
```
which results in
```
AssertionError: assert {} == None
```
unless if the parent_node field is set.

Reply:
It turns out that only two of parent_node fields were strictly necessary; I reverted/removed two of them in https://github.com/omry/omegaconf/pull/559/commits/3f40dbb0db50416ce08d659d41c1572ba31fd718.